src/match/match.ts

```diff
@@ -44,17 +44,21 @@ class Match {
     return this.allPlayers.every(player => player.connectionStatus !== ConnectionStatus.Connected);
   }
   
-  public isJoinable(matchPlayer: MatchPlayer): boolean {
-    if (matchPlayer && this.allPlayers.filter(p => p.socketId == matchPlayer.socketId).length > 0) {
-      return false;
-    }
+  public isJoinable(): boolean {
     return this.joinUntil > new Date() && this.allPlayers.length < this.maxPlayers;
   }
 
+  private removePlayer(player: MatchPlayer): void {
+    this.allPlayers.splice(this.allPlayers.findIndex(p => p == player));
+    this.playingPlayers.splice(this.playingPlayers.findIndex(p => p == player));
+    this.queueSendDataToPlayers();
+  }
+
   public addPlayer(player: MatchPlayer): boolean {
-    if (this.isJoinable(player)) {
-      if (this.allPlayers.findIndex(p => p == player) > -1) {
-        return false;
+    if (this.isJoinable()) {
+      const identicalPlayers = this.allPlayers.filter(p => p.socketId == player.socketId);
+      if (identicalPlayers.length > 0) {
+        identicalPlayers.forEach(function(p: MatchPlayer) { this.removePlayer(p); }.bind(this))
```

Comment:
Use an arrow function here to avoid explicitly binding `this` or just bind this to the removePlayer function directly. If you use an arrow function please also rename the player parameter.

Variant 1:
`identicalPlayers.forEach( (p: MatchPlayer) => this.removePlayer(p) );`

Variant 2 (preferred):
`identicalPlayers.forEach( this.removePlayer.bind(this) );`